src/H5HFman.c

```diff
@@ -487,10 +487,16 @@ H5HF__man_write(H5HF_hdr_t *hdr, const uint8_t *id, const void *obj)
     HDassert(id);
     HDassert(obj);
 
-    /* Call the internal 'op' routine routine */
-    /* (Casting away const OK - QAK) */
+    /* Call the internal 'op' routine routine
+     *
+     * In this case, the callback operation needs to modify the obj buffer that
+     * was passed in as const. We quiet the warning here because an obj pointer
+     * that was originally const should *never* arrive here.
+     */
+    H5_GCC_CLANG_DIAG_OFF("cast-qual")
     if (H5HF__man_op_real(hdr, id, H5HF__op_write, (void *)obj, H5HF_OP_MODIFY) < 0)
         HGOTO_ERROR(H5E_HEAP, H5E_CANTOPERATE, FAIL, "unable to operate on heap object")
+    H5_GCC_CLANG_DIAG_ON("cast-qual")
```

Comment:
This is quite a wordy way to de-const a pointer.

Other C projects de-const pointers without making the compiler warn by using a macro or function that casts through uintptr_t. Something like this should do the trick,

```
static inline void *
deconst(const void *p)
{
    return (void *)(uintptr_t)p;
}
```

The H5HF__man_op_real() call would turn to `H5HF__man_op_real(hdr, id, H5HF__op_write, deconst(obj), H5HF_OP_MODIFY)`.